File: backend/src/utils/api.util.ts
import type { Response } from 'express';
import type { ApiResponse } from '../types/index.js';

const VERSION = '1.0';

export function buildResponse<T>(
  success: boolean,
  data: T | null,
  message: string,
  error: string | null = null,
): ApiResponse<T> {
  return {
    success,
    data,
    message,
    error,
    meta: {
      timestamp: new Date().toISOString(),
      version: VERSION,
    },
  };
}

export function sendSuccess<T>(res: Response, data: T, message = 'OK', status = 200) {
  return res.status(status).json(buildResponse(true, data, message));
}

export function sendError(res: Response, message: string, status = 400, code?: string) {
  return res.status(status).json(
    buildResponse(false, null, message, code ?? message),
  );
}

File: backend-py/app/utils/api_util.py
from typing import Any, Optional
from fastapi.responses import JSONResponse


def build_response(success: bool, data: Any, message: str = "", error: str = "") -> dict:
    return {
        "success": success,
        "data": data,
        "message": message,
        "error": error,
        "meta": {
            "timestamp": __import__("datetime").datetime.now().isoformat(),
            "version": "1.0",
        },
    }


def send_success(data: Any, message: str = "OK", status_code: int = 200) -> JSONResponse:
    return JSONResponse(
        content=build_response(True, data, message),
        status_code=status_code,
    )
